From 4a345012ccd7326d7c51be5088da4ed5fdf8306c Mon Sep 17 00:00:00 2001
From: Andreas Cadhalpun <Andreas.Cadhalpun@googlemail.com>
Date: Mon, 25 Aug 2014 22:54:37 +0200
Subject: hardcode LLVM linker flag, because llvm-config returns only flags for
 linking statically

this reduces the size of libclamav by more than 80 percent
---
 libclamav/c++/configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libclamav/c++/configure.ac b/libclamav/c++/configure.ac
index 7fb58aa..c678991 100644
--- a/libclamav/c++/configure.ac
+++ b/libclamav/c++/configure.ac
@@ -150,6 +150,9 @@ if test "x$llvmconfig" != "x"; then
         AC_SUBST(LLVMCONFIG_LIBFILES, [`$llvmconfig --libfiles jit nativecodegen scalaropts ipo`])
     fi
 
+    AC_SUBST(LLVMCONFIG_LIBS, [-lLLVM-$llvmver])
+    AC_SUBST(LLVMCONFIG_LIBFILES, [])
+
     AC_MSG_NOTICE([CXXFLAGS from llvm-config: $LLVMCONFIG_CXXFLAGS])
     AC_MSG_NOTICE([LDFLAGS from llvm-config: $LLVMCONFIG_LDFLAGS])
     AC_MSG_NOTICE([LIBS from llvm-config: $LLVMCONFIG_LIBS])
